R/zchunk_L242.Bld_Inc_Elas_scenarios.R

```diff
@@ -34,55 +44,185 @@ module_socioeconomics_L242.Bld_Inc_Elas_scenarios_DISABLED <- function(command,
+      split(.$scenario) %>%
+      lapply(function(df) {select(df, -scenario)})
+
     # ===================================================
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+
+    L242.IncomeElasticity_bld_GCAM3 %>%
+      add_title("Building Income Elasticity: GCAM 3") %>%
+      add_units("Unitless (% change in service demand / % change in income)") %>%
+      add_comments("Calculated per-capita GDP by dividing GCAM3 pop by GCAM3 GDP") %>%
+      add_comments("Building income elasticity for each GCAM region generated by linear interpolation of assumption data") %>%
       add_legacy_name("L242.IncomeElasticity_bld_GCAM3") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("common/GCAM_region_names", "energy/A42.demand", "socioeconomics/A42.inc_elas",
+                     "L101.Pop_thous_GCAM3_R_Y","temp-data-inject/L102.gdp_mil90usd_GCAM3_R_Y") %>%
+      add_flags() ->
       L242.IncomeElasticity_bld_GCAM3
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
-      add_legacy_name("object") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
-      object
-
-    return_data(L242.IncomeElasticity_bld_GCAM3, object)
+
+    L242.pcgdp_thous90USD_Scen_R_Y[["gSSP1"]] %>%
+      add_title("Building Income Elasticity: gSSP1") %>%
+      add_units("Unitless (% change in service demand / % change in income)") %>%
+      add_comments("Uses previously calculated per-capita GDP assumptions for all SSP scenarios") %>%
+      add_comments("Building income elasticity for each GCAM region generated by linear interpolation of assumption data") %>%
+      add_legacy_name("L242.IncomeElasticity_bld_gSSP1") %>%
+      add_precursors("common/GCAM_region_names", "energy/A42.demand", "socioeconomics/A42.inc_elas",
+                     "temp-data-inject/L102.pcgdp_thous90USD_Scen_R_Y") ->
+      L242.IncomeElasticity_bld_gSSP1
+
+    L242.pcgdp_thous90USD_Scen_R_Y[["gSSP2"]] %>%
+      add_title("Building Income Elasticity: gSSP2") %>%
+      add_units("Unitless (% change in service demand / % change in income)") %>%
```

Comment:
Given how much repetition there is, does it make sense to put some of this repeated stuff in lines 94-98 above?